packages/transformer/src/IModelExporter.ts

```diff
@@ -1145,6 +1228,117 @@ export class ChangedInstanceIds {
       this.handleChange(this.element, changeType, change.ECInstanceId);
   }
 
+  /**
+   * Adds the provided change to the appropriate set of changes by class type (codeSpec, model, element, aspect) maintained by this instance of ChangedInstanceIds.
+   * If the same ECInstanceId is seen multiple times, the changedInstanceIds will be modified accordingly, i.e. if an id 'x' was updated but now we see 'x' was deleted, we will remove 'x'
+   * from the set of updatedIds and add it to the set of deletedIds for the appropriate class type.
+   * @note This method will not accept changes to relationship classes. Use 'addCustomRelationshipChange' instead.
+   * @note In most cases, this method does not need to be called. Its only for consumers to mimic changes as if they were found in a changeset, which should only be useful in certain cases such as the changing of filter criteria for a preexisting master branch relationship.
+   * @throws if the ecClassId is a relationship classId
+   * @param ecClassId class id of the custom change
+   * @param changeType insert, update or delete
+   * @param id ECInstanceID of the custom change
+   * @param federationGuid federationGuid defined on the element
+   */
+  public async addCustomChange(
+    entityType: ChangedInstanceType,
+    changeType: SqliteChangeOp,
+    id: Id64String // TODO: SUPPORT BULK ADDS (per entity type okay?)
+  ): Promise<void> {
+    if (!this._ecClassIdsInitialized) await this.setupECClassIds();
+    if (entityType === "relationship") {
+      throw new Error(
+        `Misuse. id: ${id} is a relationship. Use 'addCustomRelationshipChange' instead.`
+      );
+    }
+    this._hasCustomChanges = true;
+
+    switch (entityType) {
```

Comment:
Suggest to have multiple small methods instead of one generic one `addCustomChange`, e.g.:
- `addCustomElementChange` or `addElementChange()`
- `addCustomModelChange()` / `addModelChange()`
- `addCustomCodeSpecChange()`
- `addCustomAspectChange()`

Reply:
added small methods for each entity type